src/toxcore/dht/server/mod.rs

```diff
@@ -1492,6 +1499,10 @@ mod tests {
         pub fn has_friend(&self, pk: &PublicKey) -> bool {
             self.friends.read().contains_key(pk)
         }
+
+        pub fn add_node(&self, node: PackedNode) {
+            assert!(self.close_nodes.write().try_add(node));
```

Comment:
assert! is not a good idea since a debug version can fail in runtime and crash the app.

Reply:
`add_node` is for tests only.

Reply:
><' Sorry. github does not always show the context correctly.